rmw_zenoh_cpp/src/detail/rmw_data_types.hpp

```diff
@@ -131,4 +133,83 @@ struct rmw_subscription_data_t
   std::condition_variable * condition{nullptr};
 };
 
+
+///==============================================================================
+
+// z_owned_closure_query_t
+void service_data_handler(const z_query_t * query, void * service_data);
+
+void client_data_handler(z_owned_reply_t * reply, void * client_data);
+
+
+struct saved_queryable_data
+{
+  explicit saved_queryable_data(z_owned_query_t query)
+  : query(query)
+  {
+  }
+
+  const z_owned_query_t query;
```

Comment:
I don't think this needs to be `const`. Instead you can instantiate an `save_queryable_data` object as const.
```suggestion
  z_owned_query_t query;
```

Reply:
Done.